packages/workflow-compile/new/index.js

```diff
@@ -48,29 +46,27 @@ async function compile(config) {
 
       const compile = legacy ? shimLegacy(method) : method;
 
-      return {
-        [name]: await compile(config),
-      };
+      // return from `compile` is an array of compilations
+      return await compile(config);
     })
   );
 
   // collect results
-  //
-
-  const compilations = rawCompilations.reduce(
-    (a, b) => Object.assign({}, a, b),
-    {}
-  );
-
-  const [compilerUsed] = Object.values(compilations)
-    .map(({ compilerInfo }) => compilerInfo)
-    .filter(compilerInfo => compilerInfo);
-
-  const contracts = Object.values(compilations)
-    .map(({ contracts }) => contracts)
-    .reduce((a, b) => [...a, ...b], []);
-
-  return { contracts, compilations, compilerUsed };
+  // flatten the array and remove compilations without results
+  const compilations = rawCompilations.reduce((a, compilationGroup) => {
+    compilationGroup.forEach(compilation => {
+      if (compilation.contracts.length > 0) {
```

Comment:
So this filters out compilations where nothing was compiled, should we return them anyways? We won't have compiler information for it.

Reply:
It's not strictly correct to filter them out but I think we already do something like that and it's completely fine in practice

Reply:
Well I think in the past we've returned something like this for the Vyper compiler:
```
{ compilations: [], contracts: [], compiler: undefined }
```